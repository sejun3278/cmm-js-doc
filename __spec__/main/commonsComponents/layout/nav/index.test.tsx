import { render } from "@testing-library/react";
import LayoutNavPage from "src/main/commonsComponents/layout/nav";

export default describe("Layout Nav Page", () => {
  // 스냅샷
  test("Layout Nav Page - Snapshot", () => {
    const { container } = render(<LayoutNavPage />);
    expect(container).toMatchSnapshot();
  });

  // tap (a, p Tag)가 제대로 동작하는지 검증
  test("Layout Nav Page - Snapshot", () => {
    const { container } = render(<LayoutNavPage />);

    const listEle = container.querySelector("._layout_nav_list_");
    expect(listEle).toBeInTheDocument();

    // list가 존재하는지 검증
    if (listEle) {
      const aEle = listEle.querySelector("._tap_");
      expect(aEle).toBeInTheDocument(); // a 태그가 존재하는지 검증

      if (aEle) {
        expect(aEle).toHaveAttribute("href", "/modules/modal");
        const pEle = aEle.querySelector("._tap_name_");

        expect(pEle).toBeInTheDocument();
      }
    }
  });
});
