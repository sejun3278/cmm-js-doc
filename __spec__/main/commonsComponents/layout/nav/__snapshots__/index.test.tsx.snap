// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Nav Page Layout Nav Page - Snapshot 1`] = `
<div>
  <nav
    class="_layout_nav_wrapper_ css-1essgvo"
  >
    <article
      class="_layout_nav_list_ css-xaukxq"
    >
      <a
        class="_link_ _tap_ css-1h5x3dy"
        href="/modules/modal"
      >
        <p
          class="_p_ _tap_name_ css-pzlb5d"
        >
          Modal
        </p>
      </a>
    </article>
  </nav>
</div>
`;
